homeassistant/components/nibe_heatpump/water_heater.py

```diff
@@ -0,0 +1,182 @@
+        self._attr_device_info = coordinator.device_info
+
+        self._attr_current_operation = None
+        self._attr_target_temperature_high = None
+        self._attr_target_temperature_low = None
+        self._attr_operation_list = [STATE_HEAT_PUMP]
+
+        def _get(address: int) -> Coil:
+            return coordinator.heatpump.get_coil_by_address(address)
+
+        def _map(data: dict[str, int]) -> dict[str, Coil]:
+            return {key: _get(address) for key, address in data.items()}
+
+        self._coil_current = _get(desc.hot_water_load)
+        self._coil_start_temperature = _map(desc.start_temperature)
+        self._coil_stop_temperature = _map(desc.stop_temperature)
+        self._coil_temporary_lux: Coil | None = None
+        if desc.temporary_lux:
+            self._coil_temporary_lux = _get(desc.temporary_lux)
+        self._coil_active_accessory: Coil | None = None
+        if address := desc.active_accessory:
+            self._coil_active_accessory = _get(address)
+
+        self._coil_hot_water_comfort_mode = _get(desc.hot_water_comfort_mode)
+
+        if self._coil_temporary_lux:
+            if (
+                self._coil_temporary_lux.reverse_mappings
+                and VALUES_TEMPORARY_LUX_ONE_TIME_INCREASE
+                in self._coil_temporary_lux.reverse_mappings
+            ):
+                self._attr_operation_list.append(STATE_HIGH_DEMAND)
+            else:
+                LOGGER.warning("Skipping temporary lux since no mappings exist")
+
+        self._attr_temperature_unit = self._coil_current.unit
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if not self.coordinator.data:
+            return
+
+        def _get_float(coil: Coil | None) -> float | None:
+            if coil is None:
+                return None
+            return self.coordinator.get_coil_float(coil)
+
+        def _get_value(coil: Coil | None) -> int | str | float | None:
+            if coil is None:
+                return None
+            return self.coordinator.get_coil_value(coil)
+
+        self._attr_current_temperature = _get_float(self._coil_current)
+
+        if (mode := _get_value(self._coil_hot_water_comfort_mode)) and isinstance(
+            mode, str
+        ):
+            self._attr_target_temperature_low = _get_float(
+                self._coil_start_temperature.get(mode)
+            )
+            self._attr_target_temperature_high = _get_float(
+                self._coil_stop_temperature.get(mode)
+            )
+        else:
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+
+        if (
+            _get_value(self._coil_temporary_lux)
+            == VALUES_TEMPORARY_LUX_ONE_TIME_INCREASE
+        ):
+            self._attr_current_operation = STATE_HIGH_DEMAND
+        else:
+            self._attr_current_operation = STATE_HEAT_PUMP
+
+        self.async_write_ha_state()
```

Comment:
This works fine but personal preference for me would be to call the super method instead in case it ever changes and needs something else